core/bindings.rs

```diff
@@ -761,3 +775,57 @@ pub fn module_resolve_callback<'s>(
 
   None
 }
+
+fn get_promise_details(
+  scope: v8::FunctionCallbackScope,
+  args: v8::FunctionCallbackArguments,
+  mut rv: v8::ReturnValue,
+) {
+  if !args.get(0).is_promise() {
+    return;
```

Comment:
I'm not sure how this code is going to behave at runtime with this.  It shouldn't silently fail.  We should effectively raise a JS error with appropriate information, like a `TypeError`.

Reply:
It returned `undefined` in that way; however I like the idea of throwing `TypeError` more.